library/pod-security-policy/seccomp/template.yaml

```diff
@@ -42,45 +42,89 @@ spec:
     - target: admission.k8s.gatekeeper.sh
       rego: |
         package k8spspseccomp
-
+        
         import data.lib.exempt_container.is_exempt
-
-        violation[{"msg": msg, "details": {}}] {
-            metadata := input.review.object.metadata
-            not input_wildcard_allowed(metadata)
-            container := input_containers[_]
+        
+        container_annotation_key_prefix = "container.seccomp.security.alpha.kubernetes.io/"
+        
+        pod_annotation_key = "seccomp.security.alpha.kubernetes.io/pod"
+        
+        violation[{"msg": msg}] {
+            not input_wildcard_allowed
+            container := input_containers[name]
             not is_exempt(container)
-            not input_container_allowed(metadata, container)
-            msg := sprintf("Seccomp profile is not allowed, pod: %v, container: %v, Allowed profiles: %v", [metadata.name, container.name, input.parameters.allowedProfiles])
+            result := get_profile(container)
+            not allowed_profile(result.profile)
+            msg := sprintf("Seccomp profile '%v' is not allowed for container '%v'. Found at: %v. Allowed profiles: %v", [result.profile, name, result.location, input.parameters.allowedProfiles])
```

Comment:
should we be defensive in case input.parameters.allowedProfiles is undefined? Use object.Get() and provide an empty default?

Reply:
input.parameters.allowedProfiles is a required parameter because this policy will fail closed. If it find a profile configured and its not in the list (which it won't if empty) then a violation is triggered and pod blocked, and if it does not find a profile configured it will also trigger a violation and pod blocked.

Do we have a mechanism for requiring a parameter?

Reply:
Should be possible to set minimum list length requirements in the JSONSchema of the template